### PR TITLE
Upgrade python3 to 3.12.3

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade python3 to 3.12.3

###### Change Log  

- CVE-2024-0397
- CVE-2023-6597

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-6597
- https://nvd.nist.gov/vuln/detail/CVE-2024-0397

###### Test Methodology

- Pipeline build id: Ran full toolchain builds whose details are given in the comments section.

### CODE REVIEW

**Comment by suresh-thelkar** [comment]
I have run toolchain builds for both [amd64 ](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=605240&view=results)and [arm64 ](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=605241&view=results )which are successful.

File: SPECS/python3/python3.spec [hunk 1]
```diff
@@ -17,7 +17,6 @@ Source0:        https://www.python.org/ftp/python/%{version}/Python-%{version}.t
 # It has been removed in Python-3.12.0.tar.xz, but as our packages still require it, we will still provide for now.
 Source1:        https://github.com/python/cpython/blob/3.9/Tools/scripts/pathfix.py
 Patch0:         cgi3.patch
-Patch1:         CVE-2023-6597.patch
```
Comment (mfrw) [inline SPECS/python3/python3.spec:20]: It seems we missed to `git rm` this file.

Comment (suresh-thelkar) [reply SPECS/python3/python3.spec:20]: Thanks for catching it. I have remove the file from the git. Please check the latest commit.

**Comment by suresh-thelkar** [comment]
@mfrw, Thanks for the review. I have implemented your comments. Please take a look.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**